### PR TITLE
Quality improvements - automation

### PR DESCRIPTION
**Pull request by nCastle1**
This PR includes several classes of quality improvements, made either partially or entirely with automation (ReSharper and others). 

Note: this branch is based on the previous warnings branch.

### CODE REVIEW

File: src/Android/Xamarin.Android/Samples/Map/SearchPortalMaps/SearchPortalMaps.cs [hunk 1]
```diff
@@ -118,7 +118,7 @@ private void SearchMapsClicked(object sender, EventArgs e)
             foreach (PortalFolder folder in myContent.Folders)
             {
                 IEnumerable<PortalItem> folderItems = await portal.User.GetContentAsync(folder.FolderId);
-                mapItems.Concat(from item in folderItems where item.Type == PortalItemType.WebMap select item);
+                mapItems = mapItems.Concat(from item in folderItems where item.Type == PortalItemType.WebMap select item);
```
Comment (fragmer) [inline src/Android/Xamarin.Android/Samples/Map/SearchPortalMaps/SearchPortalMaps.cs]: Good catch! We've probably had a latent bug here for ages.

File: src/Forms/Shared/Samples/Geometry/ListTransformations/ListTransformations.xaml.cs [hunk 2]
```diff
@@ -134,7 +134,7 @@ private void GetSuitableTransformations(SpatialReference inSpatialRef, SpatialRe
             {
                 DatumTransformationListBoxItem item = new DatumTransformationListBoxItem(transform)
                 {
-                    IsDefault = (transform.Name == defaultTransform.Name)
+                    IsDefault = transform.Name == defaultTransform.Name
```
Comment (fragmer) [inline src/Forms/Shared/Samples/Geometry/ListTransformations/ListTransformations.xaml.cs]: Keep parens in cases where there is both assignment and comparison in the same statement. It's much easier to read at a glance, IMO.

File: src/UWP/ArcGISRuntime.UWP.Viewer/Samples/Geometry/BufferList/BufferList.xaml.cs [hunk 3]
```diff
@@ -108,7 +108,7 @@ public BufferList()
             try
             {
                 // Get the boolean value whether to create a single unioned buffer (true) or independent buffer around each map point (false).
-                bool unionBufferBool = (UnionCheckBox.IsChecked == true);
+                bool unionBufferBool = UnionCheckBox.IsChecked == true;
```
Comment (fragmer) [inline src/UWP/ArcGISRuntime.UWP.Viewer/Samples/Geometry/BufferList/BufferList.xaml.cs]: Keep parens.

File: src/UWP/ArcGISRuntime.UWP.Viewer/Samples/Geometry/ListTransformations/ListTransformations.xaml.cs [hunk 4]
```diff
@@ -139,7 +139,7 @@ private void GetSuitableTransformations(SpatialReference inSpatialRef, SpatialRe
             {
                 DatumTransformationListBoxItem item = new DatumTransformationListBoxItem(transform)
                 {
-                    IsDefault = (transform.Name == defaultTransform.Name)
+                    IsDefault = transform.Name == defaultTransform.Name
```
Comment (fragmer) [inline src/UWP/ArcGISRuntime.UWP.Viewer/Samples/Geometry/ListTransformations/ListTransformations.xaml.cs]: Keep parens.

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Geometry/ListTransformations/ListTransformations.xaml.cs [hunk 5]
```diff
@@ -142,7 +142,7 @@ public ListTransformations()
             {
                 DatumTransformationListBoxItem item = new DatumTransformationListBoxItem(transform)
                 {
-                    IsDefault = (transform.Name == defaultTransform.Name)
+                    IsDefault = transform.Name == defaultTransform.Name
```
Comment (fragmer) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Geometry/ListTransformations/ListTransformations.xaml.cs]: Keep parens.

File: src/iOS/Xamarin.iOS/Samples/Security/TokenSecuredChallenge/TokenSecuredChallenge.cs [hunk 6]
```diff
@@ -300,16 +300,16 @@ public LoginOverlay(CGRect frame, nfloat transparency, UIColor color, string url
             nfloat buttonWidth = 60;
 
             // Get the total height and width of the control set (five rows of controls, four sets of space).
-            nfloat totalHeight = (5 * controlHeight) + (4 * rowSpace);
+            nfloat totalHeight = 5 * controlHeight + 4 * rowSpace;
```
Comment (fragmer) [inline src/iOS/Xamarin.iOS/Samples/Security/TokenSecuredChallenge/TokenSecuredChallenge.cs]: Keep parens — code is harder to read without them.

File: src/Android/Xamarin.Android/Samples/Search/FindPlace/FindPlace.cs [hunk 7]
```diff
@@ -421,19 +421,19 @@ private void ShowStatusMessage(string message)
             string searchText = _myLocationBox.Text;
 
             // Get the results
-            IEnumerable<String> results = await GetSuggestResults(searchText);
+            List<string> results = (await GetSuggestResults(searchText)).ToList();
```
Comment (fragmer) [inline src/Android/Xamarin.Android/Samples/Search/FindPlace/FindPlace.cs]: Why not just return a List from GetSuggestResults, like we did in UWP?

Comment (nCastle1) [reply src/Android/Xamarin.Android/Samples/Search/FindPlace/FindPlace.cs]: Good catch, I meant to update all the platforms to be consistent but missed some.

**Pull request merged**